test/server/lib/HostedStorageManager.ts

```diff
@@ -1087,27 +1088,84 @@ describe('HostedStorageManager', function() {
 
   // This is a performance test, to check if the backup settings are plausible.
   describe('backupSqliteDatabase', async function() {
+
+    async function makeDb(rows: number) {
+      const tmpDir = await createTmpDir();
+      const src = path.join(tmpDir, "src.db");
+      const dest = path.join(tmpDir, "dest.db");
+      const db = await SQLiteDB.openDBRaw(src);
+      await db.run("create table data(x,y,z)");
+      await db.execTransaction(async () => {
+        const stmt = await db.prepare("INSERT INTO data VALUES (?,?,?)");
+        for (let i = 0; i < rows; i++) {
+          // Silly code to make a long random string to insert.
+          // We can make a big db faster this way.
+          const str = (new Array(100)).fill(1).map((_: any) => Math.random().toString(2)).join();
+          await stmt.run(str, str, str);
+        }
+        await stmt.finalize();
+      });
+      return {src, dest, db};
+    }
+
+    // If competing with intense user writes, backups should pause writes.
+    it(`backups will make time for themselves if competing with writes`, async function() {
+      this.timeout('10s');
+      for (const allowPause of [false, true] as const) {
+        const {db, src, dest} = await makeDb(1000);
+        let busy = 0;
+        let done = false;
+        function progress(event: BackupEvent) {
+          if (event.error && event.error.includes('SQLITE_BUSY')) {
+            busy++;
+          }
+          if (event.action === 'close' && event.phase === 'after') {
+            done = true;
+          }
+        }
+        let running = true;
+        const writerThread = (async() => {
+          while (running) {
+            // If checking without pauses, null any pause before it
+            // takes effect.
+            if (!allowPause) { db.unpause(); }
+            // Open a write transaction.
+            await db.exec('BEGIN IMMEDIATE');
+            // Hold it open a long time.
+            await delay(500);
+            // Close the write transaction.
+            await db.exec('COMMIT', { testIgnorePause: true });
```

Comment:
Why does it make sense to ignore pausing here (or elsewhere lower down in the file)?

Reply:
Here, this `COMMIT` needs to go through to end the write transaction, simulating a long write coming to an end. The logic is the same later, where I have commands I want to run to perform a manipulation on the db for the purpose of testing, the commands are not app-level writes.